go-controller/pkg/util/ovn_interface.go

```diff
@@ -0,0 +1,180 @@
+package util
+
+import (
+	goovn "github.com/ebay/go-ovn"
+)
+
+type OVNInterface interface {
```

Comment:
This interface is very large in scope. Please consider refactoring it. We could have smaller interfaces with clear responsibilities ( such as SB, NB, metrics etc)  following separation of concerns and interface segregation principles. You can compose the individual interfaces through embedding. I think this will result is clearer code, easier generating mocks etc.

Reply:
This is mimicking the go-ovn client interface: https://github.com/eBay/go-ovn/blob/master/client.go#L28.  I agree though that from a separation of concerns perspective, it's not the best. If we do decide to split it, it will mean creating more types of instances. Not sure if it helps with respect to generating mocks. The mocks are already split based on tables.

Reply:
The other implementation detail is that if we do create more types of instances for e.g. a client interface per OVN table (assuming that's what you meant), we also are increasing the number of ovndb go clients.

Reply:
Thanks @abhat for the response. My initial thinking was we have two variables for NB and SB databases.
var OVNNBDBClient OVNInterface
var OVNSBDBClient OVNInterface

If all the methods in the interface is applicable to both the databases and considering other points you said it's the best we can do. But if only subset of those calls are applicable to NB and SB databases, we can separate them into separate interfaces. We could still have a single struct OVNClient implement both the interfaces and hence there will be not much change in the code ( just the declaration of  OVNNBDBClient and OVNSBDBClient will change).
That way we will have cleaner interfaces . But if all the methods are applicable for both the databases then no point doing this.

BTW I really like you providing the extra interface in ovn-K  for go-ovn client. It'll not only able to provide the means for additional capabilities (like health check you stated), it will also allow us to swap out the underlying API if we don't like it for any reason without touching the code every where.